backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/lsm/LiveMigrateDiskCommand.java

```diff
@@ -221,64 +221,72 @@ private Set<Guid> getImageGroupIds() {
+                // in the next polling cycle
+                if (!completeLiveMigration()) {
+                    return true;
+                }
+
+                updateStage(LiveDiskMigrateStage.SOURCE_IMAGE_DELETION);
+                removeImage(getParameters().getSourceStorageDomainId(), getParameters()
+                        .getImageGroupID(), getParameters().getDestinationImageId(), AuditLogType
+                        .USER_MOVE_IMAGE_GROUP_FAILED_TO_DELETE_SRC_IMAGE);
+                updateStage(LiveDiskMigrateStage.LIVE_MIGRATE_DISK_EXEC_COMPLETED);
                 return true;
-            }
 
-            updateStage(LiveDiskMigrateStage.AUTO_GENERATED_SNAPSHOT_REMOVE_START);
-            CommandContext commandContext =
-                    ExecutionHandler.createInternalJobContext(createStepsContext(StepEnum.MERGE_SNAPSHOTS));
-            removeAutogeneratedSnapshot(commandContext, getActionType(), getParameters());
-            updateStage(LiveDiskMigrateStage.AUTO_GENERATED_SNAPSHOT_REMOVE_END);
+            case LIVE_MIGRATE_DISK_EXEC_COMPLETED:
+                if (getVm().getStatus().isQualifiedForLiveSnapshotMerge()) {
+                    // This lock is required to prevent CreateSnapshotForVmCommand from
+                    // running at the same time as RemoveSnapshotCommand
+                    //
+                    // This lock was previously acquired by the parent command, but it was released
+                    // at the end of CreateSnapshotForVm command called from the executeCommand() method.
+                    // Here the lock is acquired again and will be released when this command (LiveMigrateDisk)
+                    // finishes.
+                    if (!lockManager.acquireLock(getLock()).isAcquired()) {
+                        log.info("Failed to acquire VM lock, will retry on the next polling cycle");
+                        return true;
+                    }
+
+                    updateStage(LiveDiskMigrateStage.AUTO_GENERATED_SNAPSHOT_REMOVE_START);
+                    CommandContext commandContext =
+                            ExecutionHandler.createInternalJobContext(createStepsContext(StepEnum.MERGE_SNAPSHOTS));
+                    removeAutogeneratedSnapshot(commandContext, getActionType(), getParameters());
+                    updateStage(LiveDiskMigrateStage.AUTO_GENERATED_SNAPSHOT_REMOVE_END);
 
-            return true;
+                    return true;
+                }
+                // If the VM is not qualified for live merge, i.e. its status is not up, the auto-generated snapshot
+                // is not removed. Removing the snapshot while the VM isn't running will end up with cold merge
+                // and this is not desired here.
+                // Once cold merge enhanced to use qemu-img commit, this limit can be removed. See BZ 1246114.
+                // This behavior can be tracked by BZ 1369942.
```

Comment:
bz 1369942 was solved in 4.4 - can we update that code for cluster-levels 4.4-4.7?